test/fw/ptl/utils/pbs_logutils.py

```diff
@@ -1293,6 +1293,7 @@ def scheduler_parsing(self, rec, start, end):
         m = self.tm_tag.match(rec)
         if m:
             tm = self.logutils.convert_date_time(m.group('datetime'))
+            tm = round(tm, 0)
```

Comment:
I suggest this doing a test case that needs it. Because in my view the library should not touch the value given to it.
What if someone doesn't want to round it and use exactly what test/user-provided.
From python side, python is happy comparing float with int so if the user passes float as argument then it won't cause any problem:
```shell
$ python3
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 55.23 < 55
False
>>> 55.23 < 55.30
True
>>> 55 < 55.30
True
>>> exit()
```

Reply:
@Hiren this value is not passed from test case. This is the value that is parsed from sched logs. Are you suggesting all test cases should pass the start time and end time in microseconds from now on. We need to make changes to all those test cases.

Reply:
Assuming you are talking about `Scheduler.cycles()`:
We are already have start and end time in `cycles()` which will be passed way down to sched log analyzer and method u are changing as below:

```Scheduler.cycles() -> PBSSchedulerLog.analyse() -> (PBSLogAnalyzer.get_log_files() + ) PBSLogAnalyzer._log_parser() -> PBSLogAnalyzer.comp_analyze() -> PBSLogAnalyzer.parse_block() (+ see below) -> PBSLogUtils.in_range()```

(PBSLogAnalyzer.parse_custom_tag() nobody uses it in PTL as of now, its kinda dead code...
PBSLogAnalyzer.parse_conditional() the only pbs_loganalyzer uses it and parse_conditional() don't have to care about the start and end time as it is based on conditional/pattern matching, however, start and end time is already applied in get_log_files() so given times will be honored by log analyzer with conditional matching)

Reply:
@hirenvadalia The path for sched would be PBSSchedulerLog.comp_analyze() -> PBSSchedulerLog.scheduler_parsing if i am not wrong. from scheduler_parsing call to in_range is done.

get_log_files() checks start time and end time only with date to get the number of log files it does not parse the file for messages.

Reply:
@visheshh Yes, but even in that case also it passes start/end time in in_range which is passed by the test/user, so we should change test case, not the library.